improver/precipitation_type/snow_splitter.py

```diff
@@ -119,32 +119,38 @@ def process(self, cubes: CubeList,) -> Cube:
                         Cube of either precipitation rate or precipitation accumulation.
 
         Returns:
-            Cube of rain/snow (depending on self.variable) rate/accumulation (depending on
+            Cube of rain/snow (depending on self.output_is_rain) rate/accumulation (depending on
             precipitation cube)
+
+        Raises:
+            ValueError: If, at some grid square, both snow_cube and rain_cube have a probability of
+            0
         """
 
         rain_cube, snow_cube, precip_cube = self.separate_input_cubes(cubes)
 
         assert_spatial_coords_match([rain_cube, snow_cube, precip_cube])
+        if np.any(np.where((rain_cube + snow_cube).data == 0, True, False)):
```

Comment:
I think this will give the same result, and avoids unnecessary cube addition.
```suggestion
        if np.any((rain_cube.data + snow_cube.data) == 0):
```

Reply:
Done